src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1170,48 +1174,56 @@ public static void Reverse(ref byte buf, nuint length)
                     tempLast = Avx2.Permute2x128(tempLast, tempLast, 0b00_01);
 
                     // Store the reversed vectors
-                    tempLast.StoreUnsafe(ref buf, firstOffset);
-                    tempFirst.StoreUnsafe(ref buf, lastOffset);
-                }
-                buf = ref Unsafe.Add(ref buf, numIters * numElements);
-                length -= numIters * numElements * 2;
+                    tempLast.StoreUnsafe(ref first);
+                    tempFirst.StoreUnsafe(ref last);
+
+                    first = ref Unsafe.Add(ref first, Vector256<byte>.Count);
+                    last = ref Unsafe.Subtract(ref last, Vector256<byte>.Count);
+                } while (Unsafe.IsAddressLessThan(ref first, ref last));
```

Comment:
I steered you wrong on this initially 😅
Consider an input exactly 3 vectors long.  After the first iteration, first and last will point at the same unprocessed chunk, so it's ok to finish up with one more vector iteration.  Since there's no `Unsafe.IsAddressLessThanOrEqual`, you'll just want to flip the logic to `!Unsafe.IsAddressGreaterThan(ref first, ref last)`.

Reply:
Thanks!